app/src/main/res/layout/activity_login.xml

```diff
@@ -17,84 +17,86 @@
         />
     <TextView
         android:id="@+id/txt_user"
-        android:layout_width="100dp"
-        android:layout_height="25dp"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
         app:layout_constraintTop_toBottomOf="@id/txt_error"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintHorizontal_bias="0"
         android:text="@string/user"
         android:fontFamily="@font/roboto_bold"
         android:textColor="@color/quaternary"
-        android:textSize="16sp"
+        android:textSize="@dimen/sp16"
         />
 
     <EditText
         android:id="@+id/input_user"
         android:layout_width="match_parent"
-        android:layout_height="49dp"
+        android:layout_height="@dimen/main_activity_height_input_btn"
         android:autofillHints="name"
         android:background="@drawable/inputs_login"
-        android:inputType="text"
-        android:textSize="12sp"
-        android:textStyle="bold"
         android:fontFamily="@font/roboto_light_italic"
+        android:inputType="text"
+        android:labelFor="@id/input_user"
         android:textColor="@color/tertiary"
+        android:textSize="@dimen/sp12"
+        android:textStyle="bold"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/txt_user"
-        tools:ignore="SpeakableTextPresentCheck,TextContrastCheck" />
+        tools:ignore="SpeakableTextPresentCheck,TextContrastCheck,DuplicateSpeakableTextCheck,EditableContentDescCheck,EditableContentDescCheck" />
     <TextView
         android:id="@+id/text_errorp"
-        android:layout_width="300dp"
-        android:layout_height="25dp"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
         app:layout_constraintTop_toBottomOf="@id/input_user"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintHorizontal_bias="0"
-        android:textSize="13sp"
+        android:textSize="@dimen/sp13"
         android:fontFamily="@font/roboto_bold"
         android:textColor="#A52020"
         />
 
     <TextView
         android:id="@+id/text_pass"
-        android:layout_width="150dp"
-        android:layout_height="25dp"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
         app:layout_constraintTop_toBottomOf="@id/input_user"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        android:layout_marginTop="20dp"
+        android:layout_marginTop="@dimen/dp20"
         app:layout_constraintHorizontal_bias="0"
         android:text="@string/password"
-        android:textSize="16sp"
+        android:textSize="@dimen/main_activity_txt_size"
```

Comment:
remember not to tie dimens to specific screens when they can be reusable

Reply:
i'll implement it, thanks